tests/validator/test_resource_validator.py

```diff
@@ -0,0 +1,52 @@
+import os.path
+from unittest import TestCase
+
+from pydantic.error_wrappers import ValidationError
+from samtranslator.validator.resource_validator import to_resource_model
+from samtranslator.yaml_helper import yaml_parse
+from schema_source.aws_serverless_connector import Properties as ConnectorProperties
+
+BASE_PATH = os.path.dirname(__file__)
+CONNECTOR_INPUT_FOLDER = os.path.join(BASE_PATH, "input", "connector")
+
+
+class TestResourceModel(TestCase):
+    def setUp(self) -> None:
+        self.connector_template = {
+            "Source": {
+                "Arn": "random-arn",
+                "Type": "random-type",
+            },
+            "Destination": {"Id": "MyTable"},
+            "Permissions": ["Read"],
+        }
+
+    def test_resource_model_get(self):
+        connector_model = to_resource_model(
+            self.connector_template,
+            ConnectorProperties,
```

Comment:
I'd also have a test that's fully isolated from the rest, so we know it doesn't change, and know what it looks like (e.g. currently I'm not sure does it have nested `BaseModel`s).

Reply:
Tests added